### PR TITLE
Fix parsing comma separator for interfaces

### LINKED ISSUE #1373: Support parsing comma separator for interfaces
currently this can't parse:
```ts
interface IFoo {
    a: i32, // comma
    b: i32, // comma
}
```
or mixed:
```ts
interface IFoo {
    a: i32;
    b: i32,  // comma
    c: i32;
}
```
only `;` supports for now

### PR DESCRIPTION
**Pull request by MaxGraey**
Fix #1373

- [x] I've read the contributing guidelines

### CODE REVIEW

File: src/parser.ts [hunk 1]
```diff
@@ -2225,7 +2225,11 @@ export class Parser extends DiagnosticEmitter {
         body,
         tn.range(startPos, tn.pos)
       );
-      tn.skip(Token.SEMICOLON);
+      if (isInterface && tn.peek(false) == Token.COMMA) {
+        tn.skip(Token.COMMA);
+      } else {
+        tn.skip(Token.SEMICOLON);
+      }
```
Comment (dcodeIO) [inline src/parser.ts:2232]:
```suggestion
      if (!isInterface || !tn.skip(Token.COMMA)) {
        tn.skip(Token.SEMICOLON);
      }
```

File: src/parser.ts [hunk 2]
```diff
@@ -2169,7 +2169,7 @@ export class Parser extends DiagnosticEmitter {
             tn.range()
           );
         }
-        returnType = this.parseType(tn, name.kind == NodeKind.CONSTRUCTOR || isSetter);
+        returnType = this.parseType(tn, isSetter || name.kind == NodeKind.CONSTRUCTOR);
```
Comment (dcodeIO) [inline src/parser.ts:2172]: What's the improvement here?

Comment (dcodeIO) [reply src/parser.ts:2172]: Oh, read it the wrong way around. Nvm

Comment (MaxGraey) [reply src/parser.ts:2172]: just reorder by complexity. `isSetter` much more cheap checks in term of perf

File: src/parser.ts [hunk 3]
```diff
@@ -2302,7 +2309,11 @@ export class Parser extends DiagnosticEmitter {
         initializer,
         range
       );
-      tn.skip(Token.SEMICOLON);
+      if (isInterface && tn.peek(false) == Token.COMMA) {
+        tn.skip(Token.COMMA);
+      } else {
+        tn.skip(Token.SEMICOLON);
+      }
```
Comment (dcodeIO) [inline src/parser.ts:2316]:
```suggestion
      if (!isInterface || !tn.skip(Token.COMMA)) {
        tn.skip(Token.SEMICOLON);
      }
```

Comment (MaxGraey) [reply src/parser.ts]: Done!

File: src/parser.ts [hunk 4]
```diff
@@ -2302,7 +2307,9 @@ export class Parser extends DiagnosticEmitter {
         initializer,
         range
       );
-      tn.skip(Token.SEMICOLON);
+      if (!(isInterface && tn.skip(Token.COMMA))) {
```
Comment (MaxGraey) [inline src/parser.ts:2310]: or `!isInterface || !tn.skip(Token.COMMA)` better?

Comment (dcodeIO) [reply src/parser.ts:2310]: Looks good now!

**Pull request merged**